src/App.tsx

```diff
@@ -1,12 +1,85 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState([{
+    id: 0,
+    name: '',
+    color: '',
+  }]);
+
+  const [isErrorOccurred, setIsErrorOccured] = useState(false);
+
+  const handleShowAllButtonClick = () => {
+    try {
+      getAll().then(goods => {
+        setGoodsList(goods);
+        setIsErrorOccured(false);
+      });
+    } catch (error) {
+      setIsErrorOccured(true);
+    }
+  };
+
+  const handleShowFiveFirstButtonClick = async () => {
+    try {
+      const goods = await get5First();
+
+      setGoodsList(goods);
+      setIsErrorOccured(false);
+    } catch (error) {
+      setIsErrorOccured(true);
+    }
+  };
+
+  const handleShowRedGoodsButtonClick = async () => {
+    try {
+      const goods = await getRedGoods();
+
+      setGoodsList(goods);
+      setIsErrorOccured(false);
+    } catch (error) {
+      setIsErrorOccured(true);
+    }
+  };
```

Comment:
let's create one universal function for fetching goods (fetchGoods), it will receive a callback (getRedGoods, get5First or getAll) and call it inside and saving to state

Reply:
Refactored